src/spetlr/cosmos/cosmos_handle.py

```diff
@@ -70,7 +75,10 @@ def overwrite(
         raise NotImplementedError("Method not supported in Cosmos")
 
     def upsert(self, df: DataFrame, join_cols: List[str]) -> Union[DataFrame, None]:
-        raise NotImplementedError("Method not supported in Cosmos")
+        if set(join_cols) != set([self._id_key, self._partition_key]):
```

Comment:
set() are unordered. So the user might specify "pk, id" and gets an error.
Either make it available for users to give "pk, id" or "id, pk"
OR
ensure that the raised error gives the user instruction to set the join_cols = [id,pk]

Reply:
Sets are indeed unordered which is why the ordering does not matter, any order of id & pk will lead to equality and allow the flow to carry on. I will implement the exteded error message.